tallycash/extension
# Conflict Resolution: Introduce stub known contract and address book name resolution

---> Draft until #1304 lands.

Known contracts and the address book are stubbed in the
PreferenceService, and wired up with the expectation that they will both
eventually live in IndexedDB. Resolvers are implemented for both of
them, and used preferentially over ENS.

EIP2612 source name resolution is also moved to known contracts, making
Uniswap router resolution possible across the extension.

## Testing

- Make sure that EIP-2612 is still resolving correctly for Uniswap
  approval typed signatures.
- Check that the 0x Router is being shown instead of the full address
  on the Swap Sign Transaction page.
- Check that the 0x Router is being shown instead of the address in
  a swap transaction in the activity list.
- For a bonus, check that Uniswap is being shown in Sign Transaction
  screens for Uniswap.
